rust_snuba/rust_arroyo/src/processing/strategies/run_task_in_threads.rs

```diff
@@ -165,8 +169,17 @@ impl<TPayload, TTransformed: Send + Sync + 'static> ProcessingStrategy<TPayload>
                     Ok(Err(RunTaskError::RetryableError)) => {
                         tracing::error!("retryable error");
                     }
+                    Ok(Err(RunTaskError::Other(error))) => {
+                        // XXX: at some point we should extend the error return type of poll() to
+                        // pass those errors through to the stream processor
+                        let error: Box<dyn std::error::Error> = error.into();
+                        tracing::error!(error, "the thread errored");
+                        panic!("the thread errored");
```

Comment:
Are you actually expecting less error logging with this PR? It looks like we're still double reporting on this line (iirc this error happens on every deploy)

Reply:
we're no longer crashing the thread, so tokio at the very least should no longer report an error from within itself. i should followup with a larger refactor where we propagate the error all the way to the outside